src/share/poudriere/common.sh

```diff
@@ -3222,18 +3222,18 @@ jail_start() {
 		cp -v "${RESOLV_CONF}" "${tomnt}/etc/"
 	fi
 
-	if [ "${os}" = "CheriBSD" ]; then
-		download_toolchain_from_repo
-		download_hybridset_pkg_from_repo
-	fi
-
 	msg "Starting jail ${MASTERNAME}"
 	jstart
 	# Safe to release the lock now as jail_runs() will block further bulks.
 	slock_release "jail_start_${MASTERNAME}"
 	injail id >/dev/null 2>&1 || \
 	    err $? "Unable to execute id(1) in jail. Emulation or ABI wrong."
 
+	if [ "${os}" = "CheriBSD" ]; then
+		download_toolchain_from_repo
+		download_hybridset_pkg_from_repo
```

Comment:
Is there a reason you did them this way round? When using the jail's pkg this doesn't make sense as an order, and I don't see what the problem is doing them the other way round for the hybrid-means-host case.

Reply:
Are you asking why `download_hybridset_pkg_from_repo` is called after `download_toolchain_from_repo`? I agree it should be the other way around.